"pip install ." for Jupytext requires nodejs

We are working on a pre-commit-hook for Jupytext with @JohnPaton and @Skylion007 at #698.

I was giving a try at this `.pre-commit-config.yaml` file:
```
repos:
  - repo: https://github.com/JohnPaton/jupytext
    rev: pre-commit-hooks
    hooks:
    - id: jupytext
```
in order to test the PR.

Unfortunately, the installation of Jupytext fails with this error:
```
E                   Building wheels for collected packages: jupytext
E                     Building wheel for jupytext (PEP 517): started
E                     Building wheel for jupytext (PEP 517): finished with status 'error'
E                   Failed to build jupytext
E                   
E               stderr:
E                     ERROR: Command errored out with exit status 1:
E                      command: /home/marc/.cache/pre-commit/repoxfmha9gm/py_env-python3.8/bin/python /home/marc/.cache/pre-commit/repoxfmha9gm/py_env-python3.8/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmp8eukjuc8
E                          cwd: /tmp/pip-req-build-4fs4zglz
E                     Complete output (54 lines):
E                     running bdist_wheel
E                     running jsdeps
E                     Installing build dependencies with npm.  This may take a while...
E                     > jlpm install
E                     Traceback (most recent call last):
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/bin/jlpm", line 8, in <module>
E                         sys.exit(main())
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/jupyterlab/jlpmapp.py", line 43, in main
E                         execvp('node', ['node', YARN_PATH] + argv)
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/jupyterlab/jlpmapp.py", line 24, in execvp
E                         cmd = which(cmd)
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/jupyterlab_server/process.py", line 59, in which
E                         raise ValueError(msg)
E                     ValueError: Please install Node.js and npm before continuing installation. You may be able to install Node.js from your package manager, from conda, or directly from the Node.js website (https://nodejs.org).
E                     Traceback (most recent call last):
E                       File "/home/marc/.cache/pre-commit/repoxfmha9gm/py_env-python3.8/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
E                         main()
E                       File "/home/marc/.cache/pre-commit/repoxfmha9gm/py_env-python3.8/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
E                         json_out['return_val'] = hook(**hook_input['kwargs'])
E                       File "/home/marc/.cache/pre-commit/repoxfmha9gm/py_env-python3.8/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 204, in build_wheel
E                         return _build_backend().build_wheel(wheel_directory, config_settings,
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 216, in build_wheel
E                         return self._build_with_temp_dir(['bdist_wheel'], '.whl',
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 202, in _build_with_temp_dir
E                         self.run_setup()
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 145, in run_setup
E                         exec(compile(code, __file__, 'exec'), locals())
E                       File "setup.py", line 72, in <module>
E                         setup(
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/setuptools/__init__.py", line 153, in setup
E                         return distutils.core.setup(**attrs)
E                       File "/home/marc/miniconda3/envs/notebook_hooks/lib/python3.8/distutils/core.py", line 148, in setup
E                         dist.run_commands()
E                       File "/home/marc/miniconda3/envs/notebook_hooks/lib/python3.8/distutils/dist.py", line 966, in run_commands
E                         self.run_command(cmd)
E                       File "/home/marc/miniconda3/envs/notebook_hooks/lib/python3.8/distutils/dist.py", line 985, in run_command
E                         cmd_obj.run()
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 503, in run
E                         [self.run_command(cmd) for cmd in cmds]
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 503, in <listcomp>
E                         [self.run_command(cmd) for cmd in cmds]
E                       File "/home/marc/miniconda3/envs/notebook_hooks/lib/python3.8/distutils/cmd.py", line 313, in run_command
E                         self.distribution.run_command(command)
E                       File "/home/marc/miniconda3/envs/notebook_hooks/lib/python3.8/distutils/dist.py", line 985, in run_command
E                         cmd_obj.run()
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 274, in run
E                         c.run()
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 379, in run
E                         run(npm_cmd + ['install'], cwd=node_package)
E                       File "/tmp/pip-build-env-i5vf53rm/overlay/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 225, in run
E                         return subprocess.check_call(cmd, **kwargs)
E                       File "/home/marc/miniconda3/envs/notebook_hooks/lib/python3.8/subprocess.py", line 364, in check_call
E                         raise CalledProcessError(retcode, cmd)
E                     subprocess.CalledProcessError: Command '['/tmp/pip-build-env-i5vf53rm/overlay/bin/jlpm', 'install']' returned non-zero exit status 1.
E                     ----------------------------------------
E                     ERROR: Failed building wheel for jupytext
E                   ERROR: Could not build wheels for jupytext which use PEP 517 and cannot be installed directly
```

Another (simpler) way to reproduce the error is to try `pip install .` in the [`jupytext-dev` environment](https://github.com/mwouts/jupytext/blob/master/environment.yml).

I also note that the problem disappears when I install `nodejs` in that environment, however, I don't think we can or should take a dependency on `nodejs`! Especially in the context of pre-commit hooks, where we want to install only the CLI part of Jupytext.

--- a/jupytext/version.py
+++ b/jupytext/version.py
@@ -1,3 +1,3 @@
 """Jupytext's version number"""
 
-__version__ = "1.9.1"
+__version__ = "1.9.1+dev"

--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,9 @@
-from os import path
+from os import path, environ
 from io import open
 import re
 from setuptools import setup, find_packages
+import subprocess
+import warnings
 
 from jupyter_packaging import (
     create_cmdclass,
@@ -59,7 +61,6 @@ cmdclass = create_cmdclass(
     package_data_spec={"jupytext": ["nbextension/**"]},
     data_files_spec=data_files_spec,
 )
-
 cmdclass["jsdeps"] = combine_commands(
     install_npm(
         path.join(this_directory, "packages", "labextension"),
@@ -68,6 +69,22 @@ cmdclass["jsdeps"] = combine_commands(
     ),
     ensure_targets(jstargets),
 )
+
+# We don't build the extensions if npm is not available
+# Cf. https://github.com/mwouts/jupytext/issues/706
+try:
+    proc = subprocess.Popen(["npm", "--version"])
+except FileNotFoundError:
+    warnings.warn(
+        "The jupyterlab-jupytext extension could not be build. "
+        "Please install nodejs if you need the extensions. "
+        "See also https://github.com/mwouts/jupytext/issues/706."
+    )
+    cmdclass = {}
+
+    # We can't allow this on the CI, as we publish the package with PyPi-publish
+    if environ.get("CI"):
+        raise RuntimeError("Please install npm.")
 
 setup(
     name="jupytext",

--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ from os import path, environ
 from io import open
 import re
 from setuptools import setup, find_packages
-import subprocess
+import shutil
 import warnings
 
 from jupyter_packaging import (
@@ -72,9 +72,7 @@ cmdclass["jsdeps"] = combine_commands(
 
 # We don't build the extensions if npm is not available
 # Cf. https://github.com/mwouts/jupytext/issues/706
-try:
-    proc = subprocess.Popen(["npm", "--version"])
-except FileNotFoundError:
+if not shutil.which("npm"):
     warnings.warn(
         "The jupyterlab-jupytext extension could not be build. "
         "Please install nodejs if you need the extensions. "

